src/hyperion/device_setup_plans/adjuster.py

```diff
@@ -0,0 +1,38 @@
+from abc import ABC, abstractmethod
+from typing import Generator
+
+from bluesky import Msg
+from bluesky import plan_stubs as bps
+from ophyd import EpicsMotor
+
+from hyperion.log import LOGGER
+from hyperion.utils.lookup_table import LookupTableConverter
+
+
+class Adjuster(ABC):
+    """Abstraction that adjusts a value according to some criteria either via feedback, preset positions, lookup tables etc."""
+
+    @abstractmethod
+    def adjust(self, group=None) -> Generator[Msg, None, None]:
+        pass
+
+
+class NullAdjuster(Adjuster):
+    def adjust(self, group=None) -> Generator[Msg, None, None]:
+        pass
+
+
+class LUTAdjuster(Adjuster):
```

Comment:
Should: I think the structure of this adjuster isn't very pythonic. Unlike Java, python encourages a procedural approach first with an OOP approach taken when the problem calls for it. In this case it's not clear what other Adjusters we will need in the short/medium term so I do not think an OOP approach is required. In fact I would reduce this whole thing to a single function of something like:
```python
def adjust(lookup_table: LookupTableConverter, output_device: EpicsMotor, input):
```

Reply:
I am inclined to agree, I think it's better not to abstract until we know what all or many of the specific cases look like, and better to use loose functions than classes if they'll accomplish the same job. But I'm interested in hearing arguments for this approach?